### PR TITLE
Increase test coverage

### LINKED ISSUE #273: Improve Test Coverage
## Instructions
The current test coverage is very low so let's aad more testing to ensure the extension is as stable as possible.
- Create unit tests to thoroughly test all components
- Add integration tests for the VSCode commands

### PR DESCRIPTION
**Pull request by DylanTinianov**
Closes #273 

## Description
Setup integration tests for each layer of the extension:
1. Language Server directly
2. Emulator Controller (wraps the LS), and
3. Extension Commands (Basically e2e tests)

Added Unit tests:
- Enabled CLI parsing unit test to not require a separate parsing file 
- Added emulator key verification test
- Added tests for flow config loading key retrieval

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: extension/src/dependency-installer/installers/flow-cli-installer.ts [hunk 1]
```diff
@@ -174,3 +173,7 @@ export class InstallFlowCLI extends Installer {
     return this.checkVersion()
   }
 }
+
+export function parseFlowCliVersion (buffer: Buffer | string): string {
```
Comment (sideninja) [inline extension/src/dependency-installer/installers/flow-cli-installer.ts:177]: would be nice to have tests for it

File: extension/test/integration/2 - emulator.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { getMockSettings } from '../mock/mockSettings'
+import * as flowConfig from '../../src/emulator/local/flowConfig'
+import { Settings } from '../../src/settings/settings'
+import { MaxTimeout } from '../globals'
+import { before, after } from 'mocha'
+import * as assert from 'assert'
+import { closeTerminalEmulator, startTerminalEmulator } from './terminal-emulator'
+import { env } from 'vscode'
+
+suite('Emulator Controller', () => {
```
Comment (sideninja) [inline extension/test/integration/2 - emulator.test.ts:11]: very nice!

File: extension/test/integration/2 - emulator.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return emuCtrl.getState() === EmulatorState.Connected
+  }
+
+  async function emulatorClosed (): Promise<boolean> {
+    await emuCtrl.syncEmulatorState()
+    return emuCtrl.getState() === EmulatorState.Disconnected
+  }
+
+  test('Sync Emulator State', async () => {
+    assert.strictEqual(await startTerminalEmulator(emulatorActive, emulatorClosed), true)
```
Comment (sideninja) [inline extension/test/integration/2 - emulator.test.ts:39]: couldn't this be part of the before?

File: extension/src/dependency-installer/installers/flow-cli-installer.ts (hunk 1 above)
Comment (DylanTinianov) [reply extension/src/dependency-installer/installers/flow-cli-installer.ts:177]: There is still a test for this, just moved the function back to where it should be. But we could add some more tests as well.

File: extension/test/integration/2 - emulator.test.ts (hunk 3 above)
Comment (DylanTinianov) [reply extension/test/integration/2 - emulator.test.ts:39]: It's more of a test to make sure the emulator controller's state syncs up properly when opening and closing the emulator a few times.

**Pull request merged**